nucypher/utilities/networking.py

```diff
@@ -76,11 +80,41 @@ def __request(url: str, certificate=None) -> Union[str, None]:
         return response.text
 
 
+def _request_from_node(teacher,
+                       client: Optional[NucypherMiddlewareClient] = None,
+                       timeout: int = 2,
+                       log: Logger = IP_DETECTION_LOGGER
+                       ) -> Union[str, None]:
+    if not client:
+        client = NucypherMiddlewareClient()
+    try:
+        response = client.get(node_or_sprout=teacher, path=f"ping", timeout=timeout)  # TLS certificate logic within
+    except RestMiddleware.UnexpectedResponse:
+        # 404, 405, 500, All server response codes handled by will be caught here.
+        return  # Default teacher does not support this request - just move on.
+    except NodeSeemsToBeDown:
+        # This node is unreachable.  Move on.
+        return
+    if response.status_code == 200:
+        try:
+            ip = str(ip_address(response.text))
+        except ValueError:
+            error = f'Teacher at {teacher.rest_url()} returned an invalid IP response; Got {response.text}'
+            raise UnknownIPAddress(error)
+        log.info(f'Fetched external IP address ({ip}) from teacher ({teacher.rest_url()}).')
+        return ip
+    else:
+        # Something strange happened... move on anyways.
+        log.debug(f'Failed to get external IP from teacher node ({response.status_code})')
```

Comment:
```suggestion
        log.debug(f'Failed to get external IP from teacher node ({teacher.rest_url()} returned {response.status_code})')
```

Reply:
I'm wondering if this will still work for a `NodeSprout`?

Reply:
Interesting. Do we have that same problem for the other uses of `teacher.rest_url()` in lines above? i.e. when `response.status_code` is 200.

Reply:
:heavy_check_mark:

Reply:
Updated suggestion

```suggestion
        log.debug(f'Failed to get external IP from teacher node ({teacher} returned {response.status_code})')
```